### DISCUSSION TITLE
Q-table hanging browser

### DISCUSSION THREAD

**Discussion by CrawfordW** [quasarframework/quasar#8762] [General - Components / Directives / etc] [answered]
I have a component with a q-table dynamically populated. Haven’t touched it for several weeks, but all of a sudden, it hangs the browser as soon as any data is populated. I’ve tried reducing it to a single column, and adding one element to the data array - it still hangs the browser!
I have no idea how to resolve this - any suggestions will be welcome.

Here’s the code:
```
   <q-table
        class="sticky-header-column full-width full-height non-selectable"
        ref="table"
        :dense="dense"
        flat
        square
        virtual-scroll
        :virtual-scroll-sticky-size-start="30"
        :virtual-scroll-sticky-size-end="30"
        :rows-per-page-options="[0]"
        :pagination.sync="pagination"
        :data="tableItems"
        :columns="columns"
        row-key="name"
        selection="single"
        :selected.sync="selected"
        :sort-method="customSort"
        :filter="searchTerm"
        no-data-label="No items match filter and category setting"
    >
```

**Comment by smolinari** [comment]
Only suggestion I have is to make a minimum reproduction. Maybe someone can have a look at it.

Scott

**Comment by ontwikkelfabriek** [comment]
@CrawfordW 
>  it hangs the browser

What browser, Chrome? What happens in other browsers?

What does `hangs` mean in this case?

**Reply by CrawfordW** [reply]
Happens with Chrome and Firefox. By hang, I mean the browser becomes completely unresponsive and eventually shows a message saying something went wrong.

**Comment by hawkeye64** [comment, marked as answer]
@CrawfordW Please delete node_modules folder and your lock file (yarn.loc/package-lock.json) and re-install your packages.

**Reply by CrawfordW** [reply]
OK, did that but still have the problem.
I checked back through a bunch of commits and found the problem first arose after I installed chart.js. Do you think it messed with quasar somehow? If so, how can I uninstall chart.js and then restore quasar to it's correct state? I'm pretty green with this stuff.

**Reply by metalsadman** [reply]
> OK, did that but still have the problem.
> I checked back through a bunch of commits and found the problem first arose after I installed chart.js. Do you think it messed with quasar somehow? If so, how can I uninstall chart.js and then restore quasar to it's correct state? I'm pretty green with this stuff.

`yarn remove <package-name>`.

**Reply by CrawfordW** [reply]
I've re-built without Chart.js and the problem went away. Thanks for your help.
I've decided to build my own chart component for now. Maybe someday there will be a q-chart??

**Reply by smolinari** [reply]
@CrawfordW - Have you looked at Apex Charts?

Scott

**Reply by CrawfordW** [reply]
Scott, that looks good. I'll give it a try. Thanks.

**Reply by smolinari** [reply]
@CrawfordW https://github.com/patrickmonteiro/quasar-apexcharts

Scott

**Reply by smolinari** [reply]
😍 https://quasar-apexcharts.surge.sh/#/dashboard

Scott

**Reply by CrawfordW** [reply]
Found the problem - was error in my table props. Not related to chart.
